### PR TITLE
Add built-in pdf reader

### LINKED ISSUE #1084: Add a built in PDF reader

### PR DESCRIPTION
**Pull request by diracdeltas**
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Ran `git rebase -i` to squash commits if needed.

Fix #1084

Auditors: @bridiver @bbondy

### CODE REVIEW

File: app/extensions.js [hunk 1]
```diff
@@ -197,6 +198,13 @@ module.exports.init = () => {
   let enableExtensions = () => {
     installExtension(config.braveExtensionId, getExtensionsPath('brave'), {manifest_location: 'component', manifest: generateBraveManifest()})
 
+    if (getSetting(settings.PDFJS_ENABLED)) {
```
Comment (bridiver) [inline app/extensions.js]: do we need a setting for this? Would someone ever turn it off?

Comment (diracdeltas) [inline app/extensions.js]: PDF.js is consistently slower than just downloading the file and opening it in Preview, so some people probably won't like it even though it's built in.

Comment (BrendanEich) [inline app/extensions.js]: In general for "rich" (as in pwnable) legacy media types handled by extensions, we want a setting to disable.

Comment (bridiver) [inline app/extensions.js]: can't you still use "save link as" if you want to download? I guess I'm fine either way

**Comment by bsclifton** [comment]
This PR is beautiful :smile:

**Comment by diracdeltas** [comment]
rebased

**Comment by bbondy** [comment]
looks good to me, thanks!

**Comment by Rob--W** [comment]
Hi, PDF.js dev speaking here! It's nice to see that the extension works with minimal changes; there is only a small patch at 7880464336ef1b81545fc4c631a55bba6ad0b22a. If you explain why these changes were necessary, we can try to address the underlying issues upstream so that you don't have to separately maintain a patch.

If everything is upstream, then you could just download the CRX file from the Chrome Web store or the zip file at https://github.com/Rob--W/pdf.js/releases (those are the zip files that I upload to the Chrome Web store when I publish an update) and use it. The advantage of automatic updating is that users will then always use the latest version of PDF.js. The release process is documented here: https://github.com/mozilla/pdf.js/wiki/PDF-Viewer-(Chrome-extension)#release-process; currently the update frequence is once every few months, with some outliers to publish usability bug fixes, significant improvements and/or compatibility issues.

Feel free to open an issue upstream at https://github.com/mozilla/pdf.js/issues or mail me for questions or comments.

**Comment by diracdeltas** [comment]
@Rob--W thanks rob, good to see you here! FYI I am maintaining a fork with our patches at https://github.com/diracdeltas/pdf.js.

Our extension framework in Brave doesn't auto-update extensions so I plan to just update pdfjs every release manually with your master branch.

**Comment by CliqueBait** [comment]
Not sure if this is the right ticket but I can't seem to open any local PDFs. Online documents (e.g. http://orgmode.org/org.pdf) seem to work just fine.

Doing "File | Open File..." with Brave 0.11.1 / Ubuntu 16.04 on local copy gives me the same as this:

pasted1

**Comment by diracdeltas** [comment]
@CliqueBait that's due to https://github.com/brave/browser-laptop/issues/2714

**Pull request merged**